infrastructure/src/main/java/org/wso2/testgrid/infrastructure/providers/AWSProvider.java

```diff
@@ -304,6 +304,13 @@ private boolean doRelease(InfrastructureConfig infrastructureConfig, String stac
                 cfCompatibleParameters.add(awsParameter);
             }
 
+            if (!scriptParameter.isPresent() && expected.getParameterKey().equals("OSName")) {
```

Comment:
I think use of AMI and OSName is independent of each other. AMI is a AWS concept, while the OSName is a infrastructure_parameter property.

Reply:
Didn't get you. Could you please elaborate more?